#linux_script #課堂筆記 
---
tags: R1152筆記
---
# 07_06_2022作業修正
一共有兩個 paird end，`DRP003836`及`SRP186316`。因為之前跑 PE 的 BBMap 參數不對，導致後續 `SRP186316` 的 featureCounts 也出問題。關於為什麼`DRP003836`在 featureCounts 沒有問題，目前不確定，保險起見兩個都從 BBMap 開始重跑，以下是記錄。

至於為什麼托這麼久的原因...我在 script 中嘗試使用一些沒用過的語法，比如 `if statement` 因為等號旁沒有加空格，一整個字串會被視為 true，搞很久才找到原因，以及 `for (( i=0; i<${#numbers[@]}; i++ ))` 也花了一 些時間嘗試。最後是超時，工作階段被終止要重跑，每次重跑都要跑相當久，以上的因素使作業遲遲沒修正完成\~(>_<。)＼。~~不過它終究被完成了，希望。~~
## BBMap
### script
**script1**
原先的寫法，好像有一些問題(像是output名稱包含`_1`)，修正版本見 script2，來自秩緯。
```bash=
#!/bin/bash

cd ~/project/meta-analysis.nutrient/Arabidopsis/DRP003836/02_BBMap

INPUT_1=($(ls ./input/*_1.BBDuk-trimmed*gz))
INPUT_2=($(ls ./input/*_2.BBDuk-trimmed*gz))

[ ! -d unmapped] && mkdir unmapped
[ ! -d output.bam ] && mkdir output.bam


for ((i=0;i<=${#INPUT_1[@]};i++)); do
OUTPUT_1=`basename "${INPUT_1[i]}"| cut -f1-3 -d '.'`
OUTPUT_2=`basename "${INPUT_2[i]}"| cut -f1-3 -d '.'`


bbmap.sh -Xmx32g in="${INPUT_1[i]}" in2="${INPUT_2[i]}" ordered=t maxindel=2000 overwrite=t outm=$OUTPUT_1.sam outu=unmapped/$OUTPUT_1.unmapped.sam
samtools view -b $OUTPUT_1.sam| samtools sort -o output.bam/$OUTPUT_1.sorted.bam -
samtools index output.bam/$OUTPUT_1.sorted.bam

done

```

**script2**
```bash
#!/bin/bash

cd ~/project/meta-analysis.nutrient/Arabidopsis/DRP003836/02_BBMap
[ ! -d unmapped ] && mkdir unmapped
[ ! -d output.bam ] && mkdir output.bam


for i in `ls ./input/*1.BBDuk-trimmed*gz`; do
RUN=($(basename $i | cut -d "." -f 1-3 | cut -d "_" -f 1))
INPUT1=./input/"$RUN"_1.BBDuk-trimmed.fq.gz
INPUT2=./input/"$RUN"_2.BBDuk-trimmed.fq.gz

echo processing $RUN
bbmap.sh -Xmx32g in=$INPUT1 in2=$INPUT2 ordered=t maxindel=2000 overwrite=t outm=$RUN.sam outu=unmapped/$RUN.unmapped.sam
samtools view -b $RUN.sam| samtools sort -o output.bam/$RUN.sorted.bam -
samtools index output.bam/$RUN.sorted.bam

done
```

### result summary
result summary 的產生方式見
column 1= mapped (%)
column 2 = unambiguous (%)

**DRP003836**


**SRP186316**

## featureCounts
### script
因為只須重跑 PE，非 PE 的部分被註解掉了 
```bash=
#!/bin/bash
#dir=$(pwd)
subfol="03_featureCounts"

for folder in DRP003836 SRP007763 SRP018404 SRP075013 SRP092325 SRP186316; do
    cd ~/project/meta-analysis.nutrient/Arabidopsis/$folder/$subfol;
    echo now in $(pwd);

    [ ! -d input ] && ln -sf ../02_BBMap/output.bam/ ./input;
    ln -sf /home/hpc/ls7046-0/genome.fasta/Arabidopsis/Araport11_GTF_genes_transposons.May2022.gtf;
    [ ! -d output ] && mkdir output;

    INPUT=($(ls ./input/*bam ));
    for INPUT in ${INPUT[@]}; do
        OUTPUT=($(basename $INPUT| cut -f1-3 -d '.'));

        if [ $folder == "DRP003836" ] || [ $folder == "SRP186316" ]; then

            echo $folder PE;
            featureCounts -p --countReadPair -C -T 1 -s 0 -O -a Araport11_GTF_genes_transposons.May2022.gtf -o output/$OUTPUT.featureCounts $INPUT;

<<COMMENT
        else

            echo $folder SE;
            featureCounts -T 1 -s 0 -O -a Araport11_GTF_genes_transposons.May2022.gtf -o output/$OUTPUT.featureCounts $INPUT;
COMMENT
       fi


   done

done

#-T:thread;
#-R:output alignment result for each read, this is useful when you need to troubleshoot why were some reads not assigned;
#-s:Perform strand-specific read counting. Acceptable values:0 (unstranded), 1 (stranded) and 2 (reversely stranded).0 by default.
#-O:allowMultiOverlap,reads will be allowed to more than one matched meta-feature/gene;
#-a:annotation reference;
#-o:output;

```


### result summary
result summary 的產生方式見[29_05_2022 作業](/IfO0-cfoSxWCrr048XzaPg)
column 1 = Successfully assigned alignments
