#work_note 


# prepare R enrichment analysis input file

---

the files teacher used in R enrichment analysis
好混亂 不確定那些檔案室怎麼做的 太多了混在一起...
@ 代表 generate script **應該** 沒問題


### GWHBDNU00000000.KEGG-202207.tsv
#### format
```
GelC01G00002    K12818
GelC01G00003    K01507
GelC01G00004    K08472
```
#### generate script


### ko_definition_cleaned.txt
#### format
```
K00001  E1.1.1.1, adh; alcohol dehydrogenase [EC:1.1.1.1]
K00002  AKR1A1, adh; alcohol dehydrogenase (NADP+) [EC:1.1.1.2]
K00006  GPD1; glycerol-3-phosphate dehydrogenase (NAD+) [EC:1.1.1.8]
```



#### generate script
the original file was from [KEGG](https://www.google.com/url?q=https://www.kegg.jp/ghostkoala/&source=gmail&ust=1658211157942000&usg=AOvVaw2jOeUXS32zVOO35GUjt_-1) result `Gelata_Annotation_data.txt`
final file name is `ko_title_cleaned.tsv`

```bash
cut -d "." -f1 Gelata_Annotation_data_cleaned.txt > temp_f1.txt

cut -f2 Gelata_Annotation_data_cleaned.txt > temp_2.txt

paste temp_f1.txt temp_2.txt |sort -u > GWHBDNU00000000_KEGG.tsv

cat GWHBDNU00000000_KEGG.tsv | cut -f2 > k_number_list.txt

```


use `k_number_list.txt` to get ko title from [KO (KEGG ORTHOLOGY) Database (genome.jp)](https://www.genome.jp/kegg/ko.html)
`ko_title.txt` on my laptop


```
cat ko_title.txt | awk -v FS="   " -v RS="\n" '{print $1}' | cut -d ":" -f2

cat ko_title.txt | awk -v FS="   " -v RS="\n" '{print $1}' | cut -d ":" -f2 > temp_kof1.tsv

paste temp_kof1.tsv temp_kof2.tsv > ko_definition_cleaned.txt
```



### Gelata.GOterm-to-GDNUgene.tsv @
#### format
```
GWHGAAEX018955 GO:0004672
GWHGAAEX018955 GO:0005524
GWHGAAEX018955 GO:0006468
```

#### script
```bash
# ls7046-5@h81 ~/genome.fasta/Gastrodia/GO
awk 'FNR==NR{ a[$2]=$1; next } $1 in a{ print a[$1], $2 }' ../IDtwo_column_sort_by_GWHA.tsv temp_GWHAAEX_go_twoColumn.tsv | wc -l


awk 'FNR==NR{ a[$2]=$1; next } $1 in a{ print a[$1], $2 }' ../IDtwo_column_sort_by_GWHA.tsv temp_GWHAAEX_go_twoColumn.tsv > GWHBDNU00000000_go.tsv

perl -lane 'chomp; @l = split /\t/; for $c1(split /,/, $l[1]) { print join "\t", $l[0], $c1; }' GWHAAEX00000000_go.tsv > Gelata.GOterm-to-GDNUgene.tsv



```

### GO.term2name.tsv
#### format
```
GO:0102245      lupan-3beta,20-diol synthase activity
GO:0090707      establishment of plant organ orientation
GO:0000001      mitochondrion inheritance
```

#### script
get_GO_name.py
it works, but very slow...
```python
import urllib.request
import json
import sys

inFile = sys.argv[1]

with open(inFile, 'r') as i:
  go_term_list = i.read().splitlines()

  def get_go_term(x):
    go_term = x
    URL = f"https://www.ebi.ac.uk/QuickGO/services/ontology/go/terms/{go_term}/complete"

    response = urllib.request.urlopen(URL)
    response_json = json.loads(response.read())
    go_name = response_json['results'][0]['name']
    return go_name



  for i in (go_term_list):
     print(i, "\t", get_go_term(i))
        #print(i)
```

GO.sh
```bash
#!/bin/bash
cd ~/genome.fasta/Gastrodia/GO
python get_GO_name.py GO_term.txt > GO.term2name.tsv
```
qsub -l select=1:ncpus=1 -l walltime=10:00:00 -N "GO" -m abe -M "harewhite0@gmail.com" GO.sh

another method, [2022-07-19 GO term name type](2022-07-19%20GO%20term%20name%20type.md)